src/main/java/com/github/junrar/rarfile/FileHeader.java

```diff
@@ -159,6 +159,10 @@ public FileHeader(BlockHeader bh, byte[] fileHeader) {
                 fileName = new String(fileNameBytes);
                 fileNameW = "";
             }
+            // support Chinese characters
+            if (fileNameW.length() > 0) {
```

Comment:
what is `fileNameW` exactly?

Reply:
public boolean isUnicode() {
        return (flags & LHD_UNICODE) != 0;
    }

if LHD_UNICODE is set,the file name field contain two part.first part is the bytes of file name encoded by local charset. the second part is the bytes of file name encoded  by unicode. Two parts are separated by 0。
so i think fileNameW  represents the file name which is decoded from second part   in unicode。
String use utf-8 to decode bytes by default,so fileName garbled when the file name is Chinese